### ISSUE TITLE
Does not work with escaped characters inside regexes

### ISSUE THREAD

**Issue by netei** [eventualbuddha/decaffeinate#84]
With this file : 

```
"foo".replace(/[a-z\/]/g, "")
```
Gives the following error : 

```
~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/node_modules/magic-string/dist/magic-string.cjs.js:425
                                throw new Error('Cannot insert at replaced character index: ' + index);
                                      ^
Error: Cannot insert at replaced character index: 23
    at MagicString.insert (~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/node_modules/magic-string/dist/magic-string.cjs.js:425:11)
    at patchSemicolons (~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:1947:15)
    at ~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:4921:5
    at traverse (~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:473:23)
    at ~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:464:11
    at Array.forEach (native)
    at ~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:462:15
    at Array.forEach (native)
    at descend (~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:459:32)
    at ~/.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:4908:5
```

**Comment by jeffchan** [comment]
I believe I broke this with https://github.com/eventualbuddha/decaffeinate/pull/71 :(
There's not an easy way to distinguish `/` from escaped `\/`

**Comment by eventualbuddha** [comment]
@jeffchan looks like we just need to handle escaping properly. I'll have a PR shortly.

**Comment by eventualbuddha** [comment]
Fixed in v1.40.18.
